src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultLSPDocumentManager.cs

```diff
@@ -0,0 +1,133 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Diagnostics;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Text;
+using Microsoft.VisualStudio.Text.Editor;
+using Microsoft.VisualStudio.Threading;
+
+namespace Microsoft.VisualStudio.LanguageServerClient.Razor
+{
+    [Shared]
```

Comment:
>Shared [](start = 5, length = 6)

Why on this but not some of the other Shared ones (e.g. RazorLSPTextViewConnectionListener)? Pretty sure it's the default, right?

Reply:
Honestly it's just a habit now for MEF implementations I write with the intent being that it's defensive. That being said, if it's the default for VS' I should probably break that habit and remove them to not add in the confusion

Reply:
No worries, I don't have a problem with being very explicit, just confusing when we don't do it consistently :) 

---
In reply to: [381406922](https://github.com/dotnet/aspnetcore-tooling/pull/1602#discussion_r381406922) [](ancestors = 381406922)